docs/dependency-management.md

```diff
@@ -0,0 +1,50 @@
+# Dependency Management Guidelines
+
+This repo uses a number of third-party rust crates, so we need to be mindful
+of how these dependencies will affect our consumers.  Considerations include:
+
+* General code quality
+* [Licensing compatibility](https://www.mozilla.org/en-US/MPL/license-policy/#Licenses_Compatible_with_the_MPL)
+* Handling of security vulnerabilities
+* The potential for [supply-chain compromise](https://medium.com/intrinsic/compromised-npm-package-event-stream-d47d08605502)
+
+We're still evolving our policies in this area, but here are the
+guidelines we've developed so far:
+
+* When adding a new dependency:
+  * In general, be conservative in adding new third-party dependencies.
+    * For trivial functionality, consider just writing it yourself.
+      Remember the cautionary tale of [left-pad](https://www.theregister.co.uk/2016/03/23/npm_left_pad_chaos/).
+    * Check if we already have a crate in our dependency tree that can provide the needed functionality.
+    * Prefer widely-used crates or crates from rust-lang-nursery.
+  * Check that it is clearly licensed and is [MPL-2.0 compatible](https://www.mozilla.org/en-US/MPL/license-policy/#Licenses_Compatible_with_the_MPL).
+  * Take the time to investigate the crate's source and ensure it is suitably high-quality.
+    * Be especially wary of uses of `unsafe`.
+    * Use `cargo fetch` to obtain the same source code that cargo will eventually build.
+  * Transitively apply this reasoning to all of its dependencies.
+  * Always depend on a specific version of the dependency, avoiding semver pattern-matching.
+  * Open a separate PR just for introducing the dependency.
+    * (Editor's note: I feel like this may be controversial but wanted to put it out there for consideration...)
+    * This will help provide a clear audit trail for when and why dependencies were introduced.
+    * It will also help us remember to take these steps seriously.
+  * Dev dependencies do not require as much scrutiny as dependencies that will ship in consuming applications,
+    but should still be given some thought.
+    * There is still the potential for supply-chain compromise with dev dependencies!
+* When updating an existing dependency:
```

Comment:
I think this is the point I have the most concerns about, even though I didn't comment on my first pass. This is... fairly hard, since we don't have a diff to look at. Especially when performing this transitively.

The thing is, a lot of the time we want security or bugfixes from these deps, and so adding a lot of friction to update deps probably doesn't help us.

At the same time, I am sympathetic to the concern here, so I'm very much torn.